brave/brave-core
# Fix NFT visibility in BottomSheet fragment

## Resolves https://github.com/brave/brave-browser/issues/31915

## Details

Fixes an issue where NFTs listed in `EditVisibleAssetsBottomSheetDialogFragment` were always marked as visible.

Screen recording showing the right behavior:

https://github.com/brave/brave-core/assets/3308503/b5811252-b1e0-405d-8c0e-ac4eaa94106a

## ⚠️ Notes for Reviewers

The approach used to mark an asset as visible or invisible is going to be rewritten soon. 
The new way will leverage a WebUI based on Desktop implementation and will (probably) rely on the same persistency methods if compatible with Android. 
Because we don't want to reinvent the wheel, the actual fix is just a simple check on the NFT data model returned from core API and is implemented by the commit 👉 https://github.com/brave/brave-core/pull/20361/commits/357eaf4e0ecddd51dcbc519403f4be4db66aa01e
After the transition to a WebUI the state will be handled on the client side and not from core.

## Other Notable Changes

* `NetworkUtils#findNetwork` has been improved with a new parameter as localhost networks now can belong to FIL, SOL or ETH and on debug builds was returning wrong values.
* Improved `filterTokens` method: the filtering now uses Java streams, and it's (roughly) 50% faster as the previous logic was converting to/from array/list multiple times.
* Added Javadoc to `TokenUtils` methods.




## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

* Open Brave Wallet and navigate to NFT section by tapping on the selector to the right.
* If not present, add some NFTs by tapping on the plus icon.
* Tap on "Edit visible NFTs" and deselect all NFTs.
* Observe the full list is shown in the Hidden section.
* Tap on "Edit visible NFTs" and deselect all NFTs.
* Select all the available NFTs.
* Observe the full list is shown in the NFTs section.
* Observe the Hidden section is empty.
